download.py

```diff
@@ -36,18 +56,41 @@
     sys.stderr.write("'icann.account.password' parameter not found in the config.json file\n")
     exit(1)
 
-if not authen_base_url:
+if not authentication_base_url:
     sys.stderr.write("'authentication.base.url' parameter not found in the config.json file\n")
     exit(1)
 
 if not czds_base_url:
     sys.stderr.write("'czds.base.url' parameter not found in the config.json file\n")
     exit(1)
 
+if not aws_region:
+    sys.stderr.write("'aws.aws_region' parameter not found in the config.json file\n")
+    exit(1)
+
+if not aws_cloudwatch_namespace:
+    sys.stderr.write("'aws.cloudwatch.namespace' parameter not found in the config.json file\n")
+    exit(1)
+
 if not working_directory:
     # Default to current directory
     working_directory = '.'
 
+if (aws_access_key_id == None) ^ (aws_secret_access_key == None):
```

Comment:
Err...any reason not to just use `and` ?

I mean XOR has an X and all so immediately cool, but... :p

Reply:
I chose an XOR to avoid a number of if statements because I need to check for a condition where one of them is undefined, but not the other and that's the logical operation that best fits the bill. Generally I don't use those. :)